console/module/authorization/src/main/java/org/eclipse/kapua/app/console/module/authorization/shared/model/GwtRolePermission.java

```diff
@@ -11,10 +11,10 @@
  *******************************************************************************/
 package org.eclipse.kapua.app.console.module.authorization.shared.model;
 
-import org.eclipse.kapua.app.console.module.api.shared.model.GwtEntityModel;
+import org.eclipse.kapua.app.console.module.api.shared.model.GwtUpdatableEntityModel;
 import org.eclipse.kapua.app.console.module.authorization.shared.model.GwtPermission.GwtAction;
 
-public class GwtRolePermission extends GwtEntityModel {
+public class GwtRolePermission extends GwtUpdatableEntityModel {
```

Comment:
RolePermission is not updatable.
Please revert this.

Reply:
@Coduz The reason for this change was that when using the new `CreatedByNameCellRenderer<GwtRolePermission>()` in the `RolePermissionGrid` I got this error:

`Bound mismatch: The type GwtRolePermission is not a valid substitute for the bounded parameter <M extends GwtUpdatableEntityModel> of the type CreatedByNameCellRenderer<M>`

Without that change in GwtRolePermission class, I guess we could create a `new GridCellRenderer` in the `RolePermissionGrid` like this:
```
        columnConfig.setRenderer(new GridCellRenderer<GwtRolePermission>() {

			@Override
			public Object render(GwtRolePermission model, String property, ColumnData config, int rowIndex,
					int colIndex, ListStore<GwtRolePermission> store, Grid<GwtRolePermission> grid) {
				return model.getCreatedBy() != null ? model.getCreatedByName() : COMMONS_MSGS.notAvailable();
			}
		});
```
Would that be a better solution?

Thanks,
_Aleksandra_

Reply:
Hi @ct-ajovanovic ,

I see. The problem is that `CreatedByNameCellRenderer` is defined as:

```
public class CreatedByNameCellRenderer<M extends GwtUpdatableEntityModel> implements GridCellRenderer<M> {
```

instead of 

```
public class CreatedByNameCellRenderer<M extends GwtEntityModel> implements GridCellRenderer<M> {
```

`CreatedByNameCellRenderer` only needs the entity `GwtEntityModel` to access `getCreatedBy` and `getCreatedByName`. 

`ModifiedByNameCellRenderer` instead requires a `GwtUpdatableEntityModel` (which extends from `GwtEntityModel`)  `getModifiedBy` and `getModifiedByName`. 

Changing that should solve the problem :) 

Regards,

_Alberto_